qiskit_experiments/curve_analysis/curve_data.py

```diff
@@ -83,10 +80,36 @@ class CurveData:
     shots: np.ndarray
 
     # Maping of data index to series index
-    data_index: Union[np.ndarray, int]
+    data_allocation: np.ndarray
+
+    # List of curve names
+    labels: List[str]
 
-    # Metadata associated with each data point. Generated from the circuit metadata.
-    metadata: np.ndarray = None
+    def get_subset_of(self, index: Union[str, int]) -> "CurveData":
```

Comment:
Is this something that should be a public method? If so maybe it should be mentioned in the class docs.